### PR TITLE
opentelemetry(apm): document .NET drop in support

### PR DESCRIPTION
**Pull request by zacharycmontoya**
### What does this PR do? What is the motivation?
Documents .NET drop in support for the OpenTelemetry API.

Note: We previously had this page in our regular tracing library compatibility page so most of the content was simply moved to this new central location.

### Merge instructions
- [x] Wait until #24257 is merged, to avoid merge conflicts
- [X] Please merge after reviewing

### CODE REVIEW

**Comment by cswatt** [comment]
Tracking review with DOCS-8561

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (brett0000FF) [inline content/en/opentelemetry/interoperability/instrumentation_libraries.md:25]: All of these links are broken. I think it might be fixed by merging the latest in from master, since we made some updates for the Python and Ruby drop-in support PR.

This should be at the bottom of the page:
```
[1]: /tracing/trace_collection/
[2]: /tracing/trace_collection/automatic_instrumentation/
[3]: https://opentelemetry.io/docs/concepts/instrumentation/libraries/
[12]: /opentelemetry/interoperability/otlp_ingest_in_the_agent/?tab=host
[16]: /opentelemetry/interoperability/environment_variable_support
[24]: /getting_started/tracing/#set-up-datadog-apm
```

**Review by brett0000FF — CHANGES_REQUESTED** [review]
Thanks for adding this! I think it should be good to go once the merge conflicts are resolved from the Python and Ruby PR. Let me know if you need help working through those. 🙂

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (zacharycmontoya) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: This change should no longer be present after my rebase, closing

File: content/en/opentelemetry/interoperability/instrumentation_libraries.md
Comment (zacharycmontoya) [reply content/en/opentelemetry/interoperability/instrumentation_libraries.md]: After rebasing, these changes should no longer be present. Going to close for now

**Review by brett0000FF — APPROVED** [review]

**Pull request merged**
